 logging: avoid unnecessary printfs

Don't spend CPU cycles formatting a bunch of strings if we're
just going to throw them away later. Maybe make booting faster.

diff --git a/init/log.cpp b/init/log.cpp
index 030d2b1..13fc5fa 100644
--- a/init/log.cpp
+++ b/init/log.cpp
@@ -25,6 +25,8 @@
 static void init_klog_vwrite(int level, const char* fmt, va_list ap) {
     static const char* tag = basename(getprogname());
 
+    if (level > klog_get_level()) return;
+
     char prefix[64];
     snprintf(prefix, sizeof(prefix), "<%d>%s: ", level, tag);
 
diff --git a/libcutils/klog.c b/libcutils/klog.c
index 040d2b1..13fc5fa 100644
--- a/libcutils/klog.c
+++ b/libcutils/klog.c
@@ -62,6 +62,7 @@ void klog_writev(int level, const struct iovec* iov, int iov_count) {
 }
 
 void klog_write(int level, const char* fmt, ...) {
+    if (level > klog_level) return;
     char buf[LOG_BUF_MAX];
     va_list ap;
     va_start(ap, fmt);
